In [1]:
##Full compilation of code that works

In [9]:
import pandas as pd
df = pd.read_csv('../data/Scotland_teaching_file_1PCT(1).csv')

In [13]:
##Get list of column names
get_col = list(pd.read_csv("../data/Scotland_teaching_file_1PCT(1).csv",nrows=1).columns)
#print(get_col)
unique_values = {col: df[col].unique() for col in get_col}
print(unique_values)

{'Record_Number': array([    1,     2,     3, ..., 63386, 63387, 63388], dtype=int64), 'Region': array(['S92000003'], dtype=object), 'RESIDENCE_TYPE': array(['P', 'C'], dtype=object), 'Family_Composition': array(['1', '0', '4', '2', '5', '3', 'X'], dtype=object), 'sex': array([1, 2], dtype=int64), 'age': array([4, 3, 6, 5, 1, 2, 7, 8], dtype=int64), 'Marital_Status': array([2, 1, 4, 3, 5], dtype=int64), 'student': array([2, 1], dtype=int64), 'Country_Of_Birth': array([2, 1], dtype=int64), 'health': array([2, 3, 1, 4, 5], dtype=int64), 'Ethnic_Group': array([1, 2, 4, 3, 6, 5], dtype=int64), 'religion': array([5, 1, 2, 9, 6, 8, 4, 7, 3], dtype=int64), 'Economic_Activity': array(['1', 'X', '5', '6', '9', '4', '2', '7', '3', '8'], dtype=object), 'Occupation': array(['5', '1', '4', '2', '3', '9', 'X', '8', '6', '7'], dtype=object), 'industry': array(['5', '8', '11', '9', 'X', '2', '4', '6', '7', '10', '3', '12',
       '13', '1'], dtype=object), 'Hours_Worked_Per_Week': array(['4', '3', '2'

## Output put onto separate lines to be more legible:
### You can see the data type and that many are objects.

'Record_Number': array([    1,     2,     3, ..., 63386, 63387, 63388], dtype=int64), 

'Region': array(['S92000003'], dtype=object),

'RESIDENCE_TYPE': array(['P', 'C'], dtype=object), 

'Family_Composition': array(['1', '0', '4', '2', '5', '3', 'X'], dtype=object), 

'sex': array([1, 2], dtype=int64), 

'age': array([4, 3, 6, 5, 1, 2, 7, 8], dtype=int64), 

'Marital_Status': array([2, 1, 4, 3, 5], dtype=int64), 

'student': array([2, 1], dtype=int64), 

'Country_Of_Birth': array([2, 1], dtype=int64), 

'health': array([2, 3, 1, 4, 5], dtype=int64), 

'Ethnic_Group': array([1, 2, 4, 3, 6, 5], dtype=int64), 

'religion': array([5, 1, 2, 9, 6, 8, 4, 7, 3], dtype=int64), 

'Economic_Activity': array(['1', 'X', '5', '6', '9', '4', '2', '7', '3', '8'], dtype=object),

'Occupation': array(['5', '1', '4', '2', '3', '9', 'X', '8', '6', '7'], dtype=object), 

'industry': array(['5', '8', '11', '9', 'X', '2', '4', '6', '7', '10', '3', '12',
       '13', '1'], dtype=object), 
       
'Hours_Worked_Per_Week': array(['4', '3', '2', 'X', '1'], dtype=object),

'Approximate_Social_Grade': array(['3', '2', '4', 'X', '1'], dtype=object)}

#### For many reasons in data science it is better for objects to be strings. It would make more sense for data with a combination of numbers and the letter X to be a string. So I converted these variables to strings. 

Steps:
Type of data in column before converting
df['Column Name'].dtype

Convert column 'Column Name' from object to string

Check data type of 'Column Name' after converting to check it was successful